controllers/front/Payment.php

```diff
@@ -440,20 +482,65 @@ private function buildReturnURL($request)
      * @param $isAjax
      * @throws AdyenException
      */
-    private function processLocalPaymentMethod(\Cart $cart, $paymentType, $paymentIssuer, $isAjax)
+    private function processLocalPaymentMethod(\Cart $cart, $paymentMethod, $paymentIssuer, $isAjax)
     {
+        $isOpenInvoice = $this->paymentService->isOpenInvoicePaymentMethod($paymentMethod);
+        //$currency = CurrencyCore::getCurrencyInstance($cart->id_currency);
+
         $request = array();
         $request = $this->buildBrowserData($request);
-        $request = $this->buildLocalPaymentMethodData($paymentType, $paymentIssuer, $request);
+        $request = $this->buildLocalPaymentMethodData($paymentMethod, $paymentIssuer, $request);
         $request = $this->buildPaymentData($request);
         $request = $this->buildMerchantAccountData($request);
         $request = $this->buildReturnURL($request);
 
+
+        // Invoicing address is equal to delivery address if no invoicing address is defined during checkout
+        $invoicingAddress = new Address($cart->id_address_invoice);
+
+        $invoicingAddressCountryCode = \CountryCore::getIsoById($invoicingAddress->id_country);
+
+        if (0 != $invoicingAddress->id_state) {
+            $invoicingAddressStateIsoCode = $this->stateAdapter->getIsoById($invoicingAddress->id_state);
+        } else {
+            $invoicingAddressStateIsoCode = '';
+        }
+
+        $request = $this->addressBuilder->buildBillingAddress($invoicingAddress->address1, $invoicingAddress->address2, $invoicingAddress->postcode, $invoicingAddress->city, $invoicingAddressStateIsoCode, $invoicingAddressCountryCode, $request);
+
+
+        $deliveryAddress = new Address($cart->id_address_delivery);
+
+        $deliveryAddressCountryCode = \CountryCore::getIsoById($deliveryAddress->id_country);
+
+        if (0 != $invoicingAddress->id_state) {
+            $deliveryAddressStateIsoCode = $this->stateAdapter->getIsoById($invoicingAddress->id_state);
+        } else {
+            $deliveryAddressStateIsoCode = '';
+        }
+
+        $request = $this->addressBuilder->buildDeliveryAddress($invoicingAddress->address1, $invoicingAddress->address2, $invoicingAddress->postcode, $invoicingAddress->city, $deliveryAddressStateIsoCode, $deliveryAddressCountryCode, $request);
+
+        $customer = new \Customer($cart->id_customer);
+        $language = new \Language($cart->id_lang);
+        $localeCode = $this->languageAdapter->getLocaleCode($language);
+
+        $request = $this->customerBuilder->buildCustomerData($isOpenInvoice, $customer->email, $invoicingAddress->phone, $invoicingAddress->firstname, $invoicingAddress->lastname, $invoicingAddressCountryCode, $localeCode, $customer->id, $request);
+
+        if ($isOpenInvoice) {
+            $request = $this-> buildOpenInvoiceLines($paymentMethod, $cart, $request);
+        }
+
+        echo '<pre>';
+        var_dump($request);
+        echo '<pre>';
+        exit;
```

Comment:
oh hai

Reply:
ah remind me of old days. Can't we return it in a better way ?

Reply:
haha this is for dev only, since the logger I made is also in WIP and not even in the sprint now this is a """"nice"""" way of printing the data you need to see ;)
Please ignore it for :)